### PR TITLE
[OF-1269] feat: Tint property for gaussian splats

### PR DESCRIPTION
**Pull request by MAG-SamBirley**
The Gaussian Splat component now also describes a 'tint' property, which is intended to represent an RGB tuple that is globally applied to the Gaussian Splat asset associated with the component, when rendered in a client viewport.

### CODE REVIEW

**Comment by MAG-SamBirley** [comment]
Note that this change also introduces two new test cases to validate the behaviour of the gaussian splat component.

File: Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+#if RUN_ALL_UNIT_TESTS || RUN_GAUSSIAN_SPLAT_TESTS || RUN_GAUSSIAN_SPLAT_TEST
+CSP_PUBLIC_TEST(CSPEngine, GaussianSplatTests, GaussianSplatTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* AssetSystem	 = SystemsManager.GetAssetSystem();
+	auto* Connection					= SystemsManager.GetMultiplayerConnection();
```
Comment (MAG-AlessioRegalbuto) [inline Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp:61]: Indentation issue or git?

File: Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	EntitySystem->ProcessPendingEntityOperations();
+
+	EXPECT_EQ(GaussianSplatComponent->GetIsVisible(), true);
+
+	// Setup script
+	const std::string ScriptSource = R"xx(
+	
+		var splat = ThisEntity.getGaussianSplatComponents()[0];
+		splat.tint = [0.0, 0.1, 0.2];
+    )xx";
```
Comment (MAG-AlessioRegalbuto) [inline Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp:207]: 1 tab missing?

File: Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h [hunk 3]
```diff
@@ -124,6 +125,14 @@ class CSP_API GaussianSplatSpaceComponent : public ComponentBase,
 	/// @copydoc IShadowCasterComponent::SetIsShadowCaster()
 	void SetIsShadowCaster(bool Value) override;
 	/// @}
+
+	/// @brief Gets the tint that should be globally applied to the Gaussian Splat associated with this component.
+	/// @return The global tint value, expected to be in RGB color space.
```
[the remark this answers is not in the document]
Comment (MAG-AlessioRegalbuto) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h:130]: As a follow up question, is the vector precision double or float?

[the remark this answers is not in the document]
Comment (MAG-DanielMurdolo) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h:130]: If you check how w3c defines it, it's 0-360, 0-100%, 0-100%, so maybe we do (0-360, 0-1, 0-1)?

https://www.w3schools.com/html/html_colors_hsl.asp#:~:text=HSL%20stands%20for%20hue%2C%20saturation,an%20Alpha%20channel%20(opacity).

[the remark this answers is not in the document]
Comment (MAG-SamBirley) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h:130]: > As a follow up question, is the vector precision double or float?

> https://www.w3.org/wiki/CSS/Properties/color/HSL#:~:text=The%20format%20of%20an%20HSL,%2C%20followed

CSP Vector3's are float precision

File: Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp (hunk 1 above)
Comment (MAG-SamBirley) [reply Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp:61]: Unsure, but will investigate!

Comment (MAG-SamBirley) [reply Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp:61]: Ah this was the CSP clang formatter at work. As designed.

File: Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp (hunk 2 above)
Comment (MAG-SamBirley) [reply Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp:207]: No, more CSP clang formatter work on display. It reads fine in-IDE. As designed.

File: Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (MAG-SamBirley) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h]: @MAG-AlessioRegalbuto - latest changes include documentation that RGB values are each expected in 0..1 range. 
@MAG-DanielMurdolo - Please could you let me know if storing them as RGB and translating to HSL at the client level is OK for you.

**Review by MAG-AlessioRegalbuto — APPROVED** [review]
LGTM, nice work

File: Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (MAG-DanielMurdolo) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h]: This is the same as lights? Documentation for that doesn't say?

[the remark this answers is not in the document]
Comment (MAG-SamBirley) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h]: The documentation notes that the API expects a tuple describing a color in RGB color space, with each value normalized between 0..1 (which is consistent with how we expect light colors, and colors in general within CSP, to be treated).

I wouldn't want to start referencing the light component in the GSplat inline documentation though, as I would prefer that the GSplat documentation can stand alone.

**Review by MAG-DanielMurdolo — APPROVED** [review]

**Pull request merged**